internal/plugin/execer.go

```diff
@@ -31,6 +33,12 @@ func (c *execerClient) Implements(ctx context.Context) (bool, error) {
 
 	resp, err := c.Client.IsExecer(ctx, &empty.Empty{})
 	if err != nil {
+		// If the plugin doesn't implement IsExecer the RPC, then it definitely doesn't
+		// implement it. If we return err here, it will blow up the whole usage of this
+		// type so just say "sorry, not implemented" so the core can continue to run.
+		if st, ok := status.FromError(err); ok && st.Code() == codes.Unavailable {
```

Comment:
This is all good but an easier way to do this is `status.Code(err)` :) You can keep it like this though idc.